scripts/levitate-show-affected-plugins.js

```diff
@@ -0,0 +1,151 @@
+ * @returns {string} - The generated SQL query string.
+ */
+function makeQuery(section) {
+  const whereClause = section
+    .map(({ name, location }) => {
+      const package = getPackage(location);
+
+      if (!package) {
+        return undefined;
+      }
+
+      return `(property_name = '${name}' AND package_name = '${package}')`;
+    })
+    .filter((item) => item !== undefined)
+    .join(' OR ');
+
+  return `
+    SELECT
+      property_name,
+      package_name,
+      plugin_id
+    FROM
+      \\\`grafanalabs-global.plugins_data.plugin_imports\\\`
+    WHERE ${whereClause}
+`;
+}
+
+/**
+ * Extracts a specific column from a table represented as an array of lines.
+ *
+ * @param {Array<string>} lines - An array of strings, each representing a row in the table.
+ * @param {number} columnIndex - The index of the column to extract.
+ * @returns {Set<string>} - A set containing the unique values from the specified column.
+ */
+function getColumn(lines, columnIndex) {
+  const set = new Set();
+  const tableBody = lines.slice(3);
+
+  for (let row of tableBody) {
+    const columns = row.split('|').map((col) => col.trim());
+
+    if (columns.length === 5) {
+      const content = columns[columnIndex];
+
+      set.add(content);
+    }
+  }
+
+  return set;
+}
+
+
+/**
+ * Generates a markdown section detailing the affected plugins based on the provided data.
+ *
+ * @param {Object} data - The data object containing `removals` and `changes` arrays.
+ * @param {Array} data.removals - An array of objects representing removed items.
+ * @param {Array} data.changes - An array of objects representing changed items.
+ * @returns {string} - The generated markdown string detailing the affected plugins.
+ */
+function printAffectedPluginsSection(data) {
+  const { removals, changes } = data;
+
+  let markdown = '';
+
+  try {
+    const sqlQuery = makeQuery([...removals, ...changes]);
+    const cmd = `bq query --nouse_legacy_sql "${sqlQuery}"`;
+    const stdout = execSync(cmd, { encoding: 'utf-8' });
+
+    const rows = stdout.trim().split('\n');
+
+    if (rows.length > 3) {
+      const pluginsColumnIndex = 3;
+      const affectedPlugins = getColumn(rows, pluginsColumnIndex);
+
+      markdown += `<h3>Number of affected plugins: ${affectedPlugins.size}</h3>`;
+      markdown += "<p>To check the plugin affected by each import, click on the links below.</p>";
+
+      const propertiesColumnIndex = 1;
+      const affectingProperties = getColumn(rows, propertiesColumnIndex);
+
+      if (removals.length > 0) {
+        markdown += `<h4>Removals</h4>`;
+        markdown += createLinks(removals, affectingProperties).join('<br>\n');
+      }
+
+      if (changes.length > 0) {
+        markdown += `<h4>Changes</h4>`;
+        markdown += createLinks(changes, affectingProperties).join('<br>\n');
+      }
+    } else {
+      markdown += '<h4>No affected plugins found</h4>';
```

Comment:
What if we don't print this part and instead just output a blank report?

My concern is that this line can give a false feeling that nothing is broken and you can go ahead and merge but in truth our database of dependency usage is not 100% reliable so we can't reliable say "nothing is affected"

Reply:
Thanks for your review. It makes sense. I'll change that :D